### PR TITLE
[dev-v2.9] rancher-cis-benchmark 6.1.0-rc1 add

### PR DESCRIPTION
**Pull request by vardhaman22**
## Issue: 

https://github.com/rancher/cis-operator/issues/330
## Problem

cis-operator, kubectl and security-scan need to be bumped to latest versions
## Solution

bumped cis-operator to v1.0.14-rc4, kubectl to v1.29.7 and security-scan to v0.2.17-rc8.

### CODE REVIEW

File: charts/rancher-cis-benchmark/6.1.0-rc1/templates/scanprofilegke.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: cis.cattle.io/v1
+kind: ClusterScanProfile
+metadata:
+  name: gke-profile
+  annotations:
+    clusterscanprofile.cis.cattle.io/builtin: "true"
+spec:
+  benchmarkVersion: gke-1.2.0
```
Comment (pjbgf) [inline charts/rancher-cis-benchmark/6.1.0-rc1/templates/scanprofilegke.yml]:
```suggestion
  benchmarkVersion: gke-1.2.0

```

File: charts/rancher-cis-benchmark/6.1.0-rc1/templates/scanprofileeks.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: cis.cattle.io/v1
+kind: ClusterScanProfile
+metadata:
+  name: eks-profile
+  annotations:
+    clusterscanprofile.cis.cattle.io/builtin: "true"
+spec:
+  benchmarkVersion: eks-1.2.0
```
Comment (pjbgf) [inline charts/rancher-cis-benchmark/6.1.0-rc1/templates/scanprofileeks.yml]:
```suggestion
  benchmarkVersion: eks-1.2.0

```

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscan.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: clusterscans.cis.cattle.io
+spec:
+  group: cis.cattle.io
+  names:
+    kind: ClusterScan
+    plural: clusterscans
+  scope: Cluster
```
Comment (pjbgf) [inline charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscan.yaml]: Let's align the contents of this file with https://github.com/rancher/cis-operator/blob/master/crds/clusterscan.yaml. 

The main difference is this:

```suggestion
    plural: clusterscans
    singular: clusterscan
  scope: Cluster
```

Plus a few elements that appear in different positions. Let's just copy and paste it for the v2.9 branch.

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanbenchmark.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: clusterscanbenchmarks.cis.cattle.io
+spec:
+  group: cis.cattle.io
+  names:
+    kind: ClusterScanBenchmark
+    plural: clusterscanbenchmarks
+  scope: Cluster
```
Comment (pjbgf) [inline charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanbenchmark.yaml]: Same for this file which should match with https://github.com/rancher/cis-operator/blob/master/crds/clusterscanbenchmark.yaml.

```suggestion
    plural: clusterscanbenchmarks
    singular: clusterscanbenchmark
  scope: Cluster
```

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanprofile.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: clusterscanprofiles.cis.cattle.io
+spec:
+  group: cis.cattle.io
+  names:
+    kind: ClusterScanProfile
+    plural: clusterscanprofiles
+  scope: Cluster
```
Comment (pjbgf) [inline charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanprofile.yaml]: This file with https://github.com/rancher/cis-operator/blob/master/crds/clusterscanprofile.yaml:

```suggestion
    plural: clusterscanprofiles
    singular: clusterscanprofile
  scope: Cluster
```

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanreport.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: clusterscanreports.cis.cattle.io
+spec:
+  group: cis.cattle.io
+  names:
+    kind: ClusterScanReport
+    plural: clusterscanreports
+  scope: Cluster
```
Comment (pjbgf) [inline charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanreport.yaml]: And this with https://github.com/rancher/cis-operator/blob/master/crds/clusterscanreport.yaml.
```suggestion
    plural: clusterscanreports
    singular: clusterscanreport
  scope: Cluster
```

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanreport.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                nullable: true
+                type: string
+              lastRunTimestamp:
+                nullable: true
+                type: string
+              reportJSON:
+                nullable: true
+                type: string
+            type: object
+        type: object
```
Comment (pjbgf) [inline charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanreport.yaml]: It is a good practice to end files on a new line:
```suggestion
        type: object

```

File: charts/rancher-cis-benchmark/6.1.0-rc1/templates/validate-install-crd.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+# {{- if hasKey $found (toString .) -}}
+# 	{{- set $found (toString .) true -}}
+# {{- end -}}
+# {{- end -}}
+# {{- range $_, $exists := $found -}}
+# {{- if (eq $exists false) -}}
+# 	{{- required "Required CRDs are missing. Please install the corresponding CRD chart before installing this chart." "" -}}
+# {{- end -}}
+# {{- end -}}
+#{{- end -}}
```
Comment (pjbgf) [inline charts/rancher-cis-benchmark/6.1.0-rc1/templates/validate-install-crd.yaml:17]:
```suggestion
#{{- end -}}

```

**Review by pjbgf — CHANGES_REQUESTED** [review]

File: charts/rancher-cis-benchmark/6.1.0-rc1/templates/scanprofilegke.yml (hunk 1 above)
Comment (vardhaman22) [reply charts/rancher-cis-benchmark/6.1.0-rc1/templates/scanprofilegke.yml]: done

File: charts/rancher-cis-benchmark/6.1.0-rc1/templates/scanprofileeks.yml (hunk 2 above)
Comment (vardhaman22) [reply charts/rancher-cis-benchmark/6.1.0-rc1/templates/scanprofileeks.yml]: done

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscan.yaml (hunk 3 above)
Comment (vardhaman22) [reply charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscan.yaml]: done

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanbenchmark.yaml (hunk 4 above)
Comment (vardhaman22) [reply charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanbenchmark.yaml]: done

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanprofile.yaml (hunk 5 above)
Comment (vardhaman22) [reply charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanprofile.yaml]: done

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanreport.yaml (hunk 6 above)
Comment (vardhaman22) [reply charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanreport.yaml]: done

File: charts/rancher-cis-benchmark/6.1.0-rc1/templates/validate-install-crd.yaml (hunk 8 above)
Comment (vardhaman22) [reply charts/rancher-cis-benchmark/6.1.0-rc1/templates/validate-install-crd.yaml:17]: this file is being auto generated form charts scripts so can't modify it that's why left it as it is

File: charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanreport.yaml (hunk 7 above)
Comment (vardhaman22) [reply charts/rancher-cis-benchmark-crd/6.1.0-rc1/templates/clusterscanreport.yaml]: done..and also fixed the missing new line at end for other files as well.

**Review by pjbgf — APPROVED** [review]

**Review by mitulshah-suse — APPROVED** [review]
LGTM

**Pull request merged**